### DISCUSSION TITLE
shape_index output of aov integrator goes up with every call to render in a python session

### DISCUSSION THREAD

**Discussion by ShnitzelKiller** [mitsuba-renderer/mitsuba3#663] [Q&A] [answered]
It seems that shape_index values in the rendered map go up after every call to render(). Meaning even if I create a brand new scene, calling render will cause all the indices to be shifted up by the maximum index of the previous time I rendered with the aov integrator. Is this the intended behavior? I thought I could use this feature to get a reliable map telling me where each object in a scene is located.

**Comment by njroussel** [comment, marked as answer]
Unless you delete your previous scene, the indices will indeed just grow. 

Currently, I'm not quite sure we're able to define a consistent mapping. You'd need to turn off parallel scene loading, give names to every shape (alphanumerically increasing) and they must also be declared in the scene in that same order. Even then, I might still  be forgetting something.

We don't actually have a use case for the `shape_index` AOV, we've mostly used it as a debug view. What are you trying to achieve?

**Comment by ShnitzelKiller** [comment]
I am just trying to render a map of which pixels are covered by a particular object in randomly instantiated scenes that I am generating.

**Reply by njroussel** [reply]
Ok, I see. Apart from the recommendation I gave in my previous comment, I don't think we have a reliable way to create such a mapping. An easy solution would be to create your own tiny integrator that does exactly that by using the `si.shape` information to identify what shape you hit.
